base/abstractarray.jl

```diff
@@ -440,13 +440,15 @@ false
 checkindex(::Type{Bool}, inds::AbstractUnitRange, i) = throw(ArgumentError("unable to check bounds for indices of type $(typeof(i))"))
 checkindex(::Type{Bool}, inds::AbstractUnitRange, i::Real) = (first(inds) <= i) & (i <= last(inds))
 checkindex(::Type{Bool}, inds::AbstractUnitRange, ::Colon) = true
-checkindex(::Type{Bool}, inds::AbstractUnitRange, ::Slice) = true
+checkindex(::Type{Bool}, inds::AbstractUnitRange, s::Slice) =
+    (first(s) >= first(inds)) & (last(s) <= last(inds))
```

Comment:
Yes, that was my main concern too. `Colon` is effectively an unlimited `Slice`, which is why we never needed to check bounds before; now that it's finite, we have to check.